run_carp_tests.sh

```diff
@@ -8,11 +8,6 @@ if [[ $# -gt 0 ]] && [[ "$1" == "--no_sdl" ]]; then
     NO_SDL=1
 fi
 
-stack build;
```

Comment:
I want this line to stay...

Reply:
> I don't like the complexity of this... can't Nix use `stack` like everybody else? :)

Nix prefers pulling packages from hackage directly and curate them themselves. 

Some info on why:

https://nixos.org/nix-dev/2016-June/020585.html

IMHO nixpkgs provides a better development experience than stack. It's also great for managing non-haskell dependencies for your project.

As to whether this PR adds complexity, it does, but it would simplify my life quite a lot. Every time I switch branches I have to do some `git apply` dance to put the nix configuration in place so I can build, then I've to be careful not to commit any of those bits.

Reply:
OK, interesting. I'll test your PR on my machines and get back after some more evaluation.